BOJ/6588_김덕래.cpp

```diff
@@ -0,0 +1,55 @@
+#include<iostream>
+#include<algorithm>
+#include<vector>
+#define MAX 1000000
+using namespace std;
+
+bool nonPrimeCheck[MAX + 1];
+vector<int> test;
+vector<int> prime;
+void input();
+void eratos();
+void solve();
+
+int main(){
+    cin.tie(NULL);
+    ios_base::sync_with_stdio(false);
+    input();
+    solve();
+    return 0;
+}
+
+void input(){
+    int n;
+    do{
+        cin >> n;
+        test.push_back(n);
+    }while(n);
+    test.pop_back();
```

Comment:
```suggestion
    while(1){
        cin >> n;
        if(n == 0) break;
        test.push_back(n);
    }
```
굳이 n=0일 때 test에 값을 넣고 pop_back을 할 필요가 없을 거 같아요!